### ISSUE TITLE
DAQ: Overvoltage and undervoltage alert debouncing calculation wrong if sampling frequency != 1kHz

### ISSUE THREAD

**Issue by martinjaeger** [LibreSolar/charge-controller-firmware#109]
The debouncing depends on the sampling frequency being 1 kHz, so one call is equivalent to 1 millisecond. This should be fixed to work properly in STM32G4 based charge controllers which currently run the ADC at much higher sampling rate.

**Comment by JordanSilverman** [comment]
Do you think this was the issue that caused the overvoltage protection to fail in your controller?

**Comment by martinjaeger** [comment]
No, don't think so, as it was only configured for battery overvoltage so far. But I realized this issue while implementing the protection for the high-side aswell.
